### DISCUSSION TITLE
Required field throws error when using controlled state

### DISCUSSION THREAD

**Discussion by BennyAlex** [lookfirst/mui-rff#1027] [General]
Hi,

For some fields, I want the value to be controlled, eg. by state or context.
Therefore I set value prop and onChange by myself.
But this gives me now an error when using this in conjunction with yup.required.

Here is an example: https://codesandbox.io/s/react-final-form-material-ui-example-forked-hym8ci?file=/src/index.tsx

When typing a value in obj.id field, it still shows the required error. Removing the controlled state (value & onChange) it works as expected.

**Comment by lookfirst** [comment]
Hi there, it is not clear to me what you're referring to at this point. Please provide a minimal example.

**Reply by BennyAlex** [reply]
Yes I did:
![image](https://user-images.githubusercontent.com/17402724/225151156-57eaf17b-7916-4e65-ad88-1e757bc8d7a2.png)

Typing in this field still throws the required error, because I controll its state by React.useState
